systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomAuthorizerST.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.security.custom;
+
+
+import io.strimzi.api.kafka.model.AclOperation;
+import io.strimzi.api.kafka.model.KafkaAuthorizationSimple;
+import io.strimzi.api.kafka.model.listener.KafkaListenerAuthenticationTls;
+import io.strimzi.api.kafka.model.listener.arraylistener.KafkaListenerType;
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.kafkaclients.externalClients.BasicExternalKafkaClient;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.resources.crd.KafkaTopicResource;
+import io.strimzi.systemtest.resources.crd.KafkaUserResource;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.test.WaitException;
+
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.systemtest.Constants.EXTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.NODEPORT_SUPPORTED;
+
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@Tag(REGRESSION)
+public class CustomAuthorizerST extends AbstractST {
+    static final String NAMESPACE = "custom-authorizer-test";
+    static final String CLUSTER_NAME = "custom-authorizer";
+    static final String ADMIN = "sre-admin";
+    private static final Logger LOGGER = LogManager.getLogger(CustomAuthorizerST.class);
+
+    @Test
+    @Tag(NODEPORT_SUPPORTED)
+    @Tag(EXTERNAL_CLIENTS_USED)
```

Comment:
@Frawless @im-konge I do not think these tests really need to use external clients, or? Can you help @rareddy use internal clients for this?

Reply:
@rareddy  could you please check test `testSendMessagesTlsAuthenticated` how we use `InternalKafkaClient` ? I think that simply replace of `BasicExternalKafkaClient` exactly with the code form test I mentioned should be enough.

Reply:
Thanks, @Frawless & @scholzj  updated test to the internal client now.